### ISSUE TITLE
Change registering as an outgoing tx

### ISSUE THREAD

**Issue by Fonta1n3** [cryptoadvance/specter-desktop#247]
Looks like a change output is showing as an outgoing tx and has deducted from my balance so that my balance is no longer accurate, I would share the txid etc but unfortunately its all mainnet.

I am using the same wallet across multiple apps, it is a 2 of 3 m/48'/0'/0'/2' derivation.

Does Specter not handle change keys as per "BIP48" with `m/48'/0'/0'/2'/1`? I had assumed if I created a `m/48'/0'/0'/2'` wallet that it would import receive and change keys with `m/48'/0'/0'/2'/0` and `m/48'/0'/0'/2'/1` respectively. I could have made an incorrect assumption about "BIP48" internal path.

FWIW Fully Noded automatically imports the receive and internal paths as per the above, so if a user exports a Specter wallet the change outputs will show as outgoing tx in Specter which will probably scare people. I had created this wallet on Specter a couple months ago perhaps this has been updated since then.

I checked the Specter wallet on my node directly and can verify it is also showing the same. The only explanation would be that the change keys were not imported or I misunderstood how change is derived with the 48 derivation (we really need an official BIP for that).

**Comment by stepansnigirev** [comment]
It should add `m/48'/0'/0'/2'/1/*` as change descriptors as well... Are you using `sortedmulti` or `multi` descriptor?

**Comment by Fonta1n3** [comment]
Externally to Specter its `sortedmulti`

**Comment by Fonta1n3** [comment]
Just recreated the wallet and rescanned, same result.

**Comment by ben-kaufman** [comment]
@Fonta1n3 if this is still an issue, can you please share the wallet json file, just deleting the personal info? (so just in xpubs and such delete everything but the `xpub` prefix and so on). You could find that undder the `~/.specter/wallets/` folder.

**Comment by Fonta1n3** [comment]
{
    "name": "Multisig",
    "alias": "multisig",
    "description": "2 of 3 Multisig (Segwit)",
    "address_type": "bech32",
    "address": "bc1qq6qt06u....rrl9d6pqjxc6jqet0ygc",
    "address_index": 12,
    "change_address": "bc1q6cxr.....76fflqptxlqk0tjg4",
    "change_index": 4,
    "keypool": 32,
    "change_keypool": 24,
    "recv_descriptor": "wsh(sortedmulti(2,[73756c7f/48h/0h/0h/2h]xpub6E3pgkahcxomzJRV.....bsAiN4W86bKfpihD/0/*,[f9f62194/48h/0h/0h/2h]xpub6EuULBfjrx1sHZj9vYFbAQ.....zXn9W56xSoeft551jhcgJV8NzsJ6k3i/0/*,[c98b1535/48h/0h/0h/2h]xpub6EmHFhyZkMjCZ......ULBwHQ29LNVwB7mMq/0/*))#tjkvyrcd",
    "change_descriptor": "wsh(sortedmulti(2,[73756c7f/48h/0h/0h/2h]xpub6E3pgkahcxomzJRV......86bKfpihD/1/*,[f9f62194/48h/0h/0h/2h]xpub6EuULBfjrx1sHZj9vYFb......xSoeft551jhcgJV8NzsJ6k3i/1/*,[c98b1535/48h/0h/0h/2h]xpub6EmHFhyZk.....mt8Z5ULBwHQ29LNVwB7mMq/1/*))#w8ak69g9",
    "keys": [
        {
            "original": "xpub6E3pgkahcxomz......dgEnfzcGWVgbsAiN4W86bKfpihD",
            "fingerprint": "73756c7f",
            "derivation": "m/48h/0h/0h/2h",
            "type": "wsh",
            "xpub": "xpub6E3pgka....zcGWVgbsAiN4W86bKfpihD"
        },
        {
            "original": "xpub6EuULBfjrx1s.......51jhcgJV8NzsJ6k3i",
            "fingerprint": "f9f62194",
            "derivation": "m/48h/0h/0h/2h",
            "type": "wsh",
            "xpub": "xpub6EuULBfjrx1sHZj9vYFbAQXR....dJCfpjGAH72B13CRrpFcpQzXn9W56xSoeft551jhcgJV8NzsJ6k3i"
        },
        {
            "original": "xpub6EmHFhyZkMjCZLy3vrfjWq....XySZTMCcKGdwFs6Eqz5opC6emt8Z5ULBwHQ29LNVwB7mMq",
            "fingerprint": "c98b1535",
            "derivation": "m/48h/0h/0h/2h",
            "type": "wsh",
            "xpub": "xpub6EmHFhyZkMjCZLy3vrfjWqPCo8CT.....cKGdwFs6Eqz5opC6emt8Z5ULBwHQ29LNVwB7mMq"
        }
    ],
    "devices": [
        "coldcard_2of3",
        "iphone",
        "electrum"
    ],
    "sigs_required": 2,
    "pending_psbts": {},
    "fullpath": "/Users/peter/.specter/wallets/main/multisig.json"
}

**Comment by stepansnigirev** [comment]
And what is the derivation path of the change address that Specter doesn't recognize?

**Comment by Fonta1n3** [comment]
Here is the censored `bitcoin-cli getaddressinfo` result for the change address, the issue is the `change_keypool` is `24` whereas the change address is at `26`.

```
{
    address = bc1qxflufqvvr6e......aqpwsj20sg7uphv;
    desc = "wsh(multi(2,[c98b1535/48'/0'/0'/2'/1/26]0280193a7......2ce7712c4,[f9f62194/48'/0'/0'/2'/1/26]03af1ff7c8a......6c69e40003432,[73756c7f/48'/0'/0'/2'/1/26]03e28354.....5bc587c28939))#xlfczdjj";
}
```

**Comment by Fonta1n3** [comment]
I swear I add 100 more addresses previously... Adding more addresses has solved the issue.

**Comment by stepansnigirev** [comment]
Maybe it didn't work for some reason. I think we need to carefully look at this part of Specter and make sure it watches addresses properly. We didn't really test the scenario when transactions are made by external wallet.

**Comment by Fonta1n3** [comment]
I am sure when I added more addresses previously it was not on the current version of Specter, in any case it is working fine. This can definitely be closed. I am curious though as to why Specter only watches 20 keys initially? If it is issuing commands locally to the node I would have assumed you would just import a ~thousand keys and be done with it?

**Comment by stepansnigirev** [comment]
The reason is that Bitcoin Core v19 doesn't support sortedmulti, so we had to import every address individually. And also getting the public keys from descriptorinfo call, so 2 rpc calls per address, 20 change and 20 receiving addresses - 80 rpc calls in total. For 1000 keys the user would need to wait extremely long time for all addresses to import. It was a tradeoff we had to make :(
This is likely to change in one of upcoming releases, we could bump the gap limit to ~100 at least.
Now we grouped the calls and will continue optimizations, so we could
